### ISSUE TITLE
[Question] Does the waiting time during retry count for billing?

### ISSUE THREAD

**Issue by MhAllan** [Azure/Azure-Functions#2122]
I am asking here because there is no documentations about this.

do I get charged for the time the function is waiting to retry when I define retry policy like FixedDelayRetry or ExponentialDelayRetry?

**Comment by v-bbalaiagar** [comment]
Hi @MhAllan, Thank you for your feedback! We will check for the possibilities internally and update you with the findings.

**Comment by v-bbalaiagar** [comment]
Hi @divyagandhisethi, Do you have context to answer this

**Comment by v-bbalaiagar** [comment]
Hi @MhAllan , I checked this internally. Host does not emit any logs that will be used to create billing meters on the platform but worker instance continues to live if retries are in progress. So we will not be charging Cx for retries. 

cc @fabiocav

**Comment by MhAllan** [comment]
@v-bbalaiagar thanks for the reply, but sorry I didn't get the answer to my question, I am asking if the time between retries is added to the execution time billing. I am not asking about the count of retries, but the time or retries.

**Comment by v-bbalaiagar** [comment]
Tagging @AnatoliB  for further insights

**Comment by AnatoliB** [comment]
The time _between_ retries is _not_ added to the execution time billing.
cc @eamonoreilly for documentation update, if necessary.

**Comment by MhAllan** [comment]
@AnatoliB , thanks for your reply

**Comment by v-bbalaiagar** [comment]
Hi @MhAllan , Closing this issue as the new issue opened here - https://github.com/MicrosoftDocs/azure-docs/issues/88818

**Comment by MhAllan** [comment]
Hi @AnatoliB, sorry to reach you again here but the documentation is really not helping. so according to this discussion here and according to this documentation [here](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-error-pages?tabs=csharp#retry-policy-options). 

> The current retry count is stored in memory of the instance. 

So if retry policy information is stored in memory, does that mean if I have 1 week delay for retry, my function will stay in memory for 1 week without paying? I am not getting it.

**Comment by AnatoliB** [comment]
@MhAllan Retry delay duration is limited, I doubt you can specify 1 week. If you need to pause for this long, consider using Durable Functions (e.g. https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-overview?tabs=csharp#monitoring).

@alrod Do we have an explicit limit on the delay duration, or on the total execution time including retries? Is it documented anywhere?

**Comment by MhAllan** [comment]
@AnatoliB I am asking because I want to replace durable functions with RetryPolicies If I can do that =)

**Comment by AnatoliB** [comment]
> @AnatoliB I am asking because I want to replace durable functions with RetryPolicies If I can do that =)

@MhAllan That was unexpected. :-) If you want to wait longer than 5-10 minutes between retries, durable functions is a much more robust solution. As you already know, retries with the Retry Policy feature will work only if the host process is still around: we will not keep it around for days just for the purposes of retrying, and we will not spawn another one just for the purposes of retrying.

**Comment by MhAllan** [comment]
@AnatoliB  yeah you are right, except my use case looks cleaner without managing orchestrators and sub-orchestrators

**Comment by ggailey777** [comment]
I'm adding a reminder about only billing for execution time (and not retry wait time) in a Consumption plan to the article.
